app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,65 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, only: %i[edit update]
+  before_action :valid_user, only: %i[edit update]
+  before_action :check_expiration, only: %i[edit update]
+
+  def new; end
+
+  def create
+    @user = User.find_by(email: params[:password_reset][:email].downcase)
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = 'Email sent with password reset instructions'
+      redirect_to root_url
+    else
+      # flash doesn't work, FIX it!
+      # flash.now[:danger] = 'Email address not found'
+      flash[:danger] = 'Email address not found'
+      render 'new'
+    end
+  end
+
+  def update
+    if params[:user][:password].empty?
```

Comment:
Also, I'ld rather base on user model validations here where user password presence should exist